### PR TITLE
zh-TW: cr2 Day 1 translations

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all zh-TW content is reviewed.

#1463 #684

### CODE REVIEW

File: po/zh-TW.po [hunk 1]
```diff
@@ -2509,6 +2513,8 @@ msgid ""
 "should have immediate parallels in other languages. The more advanced parts "
 "of Rust come on the subsequent days."
 msgstr ""
+"第一天的規畫是說明 Rust 的「基礎」概念，這應直接對應到其他語言中的相似概念。"
```
Comment (mingyc) [inline po/zh-TW.po:2516]:
```suggestion
"第一天的規畫是說明 Rust 中能夠直接對應到其他語言的「基礎」概念。"
```

File: po/zh-TW.po [hunk 2]
```diff
@@ -2509,6 +2513,8 @@ msgid ""
...
+"後續幾天則會介紹更進階的 Rust 程式語言。"
```
Comment (mingyc) [inline po/zh-TW.po:2517]:
```suggestion
"後續幾天則會介紹更進階的部分。"
```

File: po/zh-TW.po [hunk 3]
```diff
@@ -920,7 +906,7 @@ msgstr "範例"
 
 #: src/SUMMARY.md src/chromium/interoperability-with-cpp/limitations-of-cxx.md
 msgid "Limitations of CXX"
-msgstr ""
+msgstr "CXX 限制"
```
Comment (mingyc) [inline po/zh-TW.po:909]:
```suggestion
msgstr "CXX 的限制"
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -2634,7 +2643,7 @@ msgstr "Rust 的幾個獨特賣點如下："
 msgid ""
 "_Compile time memory safety_ - whole classes of memory bugs are prevented at "
 "compile time"
-msgstr ""
+msgstr "__確保編譯期間的記憶體安全性 - 在編譯期間避免各類記憶體錯誤"
```
Comment (mingyc) [inline po/zh-TW.po:2646]:
```suggestion
msgstr "__編譯期的記憶體安全性 - 在編譯期間就能避免各類記憶體錯誤"
```

File: po/zh-TW.po [hunk 5]
```diff
@@ -2668,7 +2677,7 @@ msgstr "不會發生疊代器無效的情形。"
 msgid ""
 "_No undefined runtime behavior_ - what a Rust statement does is never left "
 "unspecified"
-msgstr ""
+msgstr "__不會出現未定義的執行階段行為 - Rust 陳述式的行為一律會經過指定"
```
Comment (mingyc) [inline po/zh-TW.po:2680]:
```suggestion
msgstr "__不會出現未定義的執行階段行為 (undefined runtime behavior) - Rust 陳述式的行為一律會有定義"
```

File: po/zh-TW.po [hunk 6]
```diff
@@ -2759,25 +2768,31 @@ msgid ""
...
+"[Rust Playground](https://play.rust-lang.org/) 支援以簡便方式執行精簡的 Rust "
+"程式，也是本課程中範例和練習的基礎。不妨試著執行 Rust Playground 開頭的"
+"「hello-world」程式。Playground 兼具以下幾項便利功能："
 
 #: src/hello-world/playground.md
 msgid ""
 "Under \"Tools\", use the `rustfmt` option to format your code in the "
 "\"standard\" way."
-msgstr ""
+msgstr "在「Tools」下方點選「rustfmt」選項，以「標準」方式設定程式碼格式。"
```
Comment (mingyc) [inline po/zh-TW.po:2779]:
```suggestion
msgstr "在「Tools」下方點選「`rustfmt`」選項，以「標準」方式設定程式碼格式。"
```

File: po/zh-TW.po [hunk 7]
```diff
@@ -3045,25 +3071,29 @@ msgstr ""
...
-msgstr ""
+msgstr "在其他語言中，算數的方法非常相似，優先順序也雷同。"
 
 #: src/types-and-values/arithmetic.md
 msgid ""
 "What about integer overflow? In C and C++ overflow of _signed_ integers is "
 "actually undefined, and might do different things on different platforms or "
 "compilers. In Rust, it's defined."
 msgstr ""
+"那麼整數溢位現象呢？在 C 和 C++ 中，__「帶號」整數的溢位現象實際上並未定義，"
```
Comment (mingyc) [inline po/zh-TW.po:3095]:
```suggestion
"那麼整數溢位現象呢？在 C 和 C++ 中，「帶號」整數的溢位現象實際上並未定義，"
```

File: po/zh-TW.po [hunk 8]
```diff
@@ -3073,18 +3103,25 @@ msgid ""
...
 "why the example requires a separate function."
 msgstr ""
+"事實上，編譯器會偵測常數運算式的溢位，這也是本例中需要另一個函式的原因。"
 
 #: src/types-and-values/strings.md
 msgid ""
 "Rust has two types to represent strings, both of which will be covered in "
 "more depth later. Both _always_ store UTF-8 encoded strings."
 msgstr ""
+"Rust 用來代表字串的型別有兩種，稍後會深入介紹。兩者__「一律」都儲存 UTF-8 編"
```
Comment (mingyc) [inline po/zh-TW.po:3123]:
```suggestion
"Rust 用來代表字串的型別有兩種，稍後會深入介紹。兩者「一律」都儲存 UTF-8 編"
```

**Review by mingyc — CHANGES_REQUESTED** [review]

**Comment by henrif75** [comment]
Hi @mingyc thanks for your review!
The CLA check is failing probably because you have used your personal email, can you check whether you have it associated to your corp account?

**Comment by mingyc** [comment]
Weird... It used to pass the check as I always use this address... Anyway it should be resolved now.

File: po/zh-TW.po [hunk 9]
```diff
@@ -939,85 +924,85 @@ msgstr "錯誤處理"
...
 
 #: src/SUMMARY.md
 msgid "Reviews and Audits"
-msgstr ""
+msgstr "審查和稽核"
 
 #: src/SUMMARY.md
 msgid "Checking into Chromium Source Code"
-msgstr ""
+msgstr "登錄為 Chromium 原始碼"
```
Comment (mingyc) [inline po/zh-TW.po:976]:
```suggestion
msgstr "提交為 Chromium 原始碼"
```

File: po/zh-TW.po [hunk 10]
```diff
@@ -939,85 +924,85 @@ msgstr "錯誤處理"
...
 
 #: src/SUMMARY.md src/chromium/adding-third-party-crates/keeping-up-to-date.md
 msgid "Keeping Crates Up to Date"
-msgstr ""
+msgstr "讓 Crate 保持最新狀態"
```
Comment (mingyc) [inline po/zh-TW.po:980]:
```suggestion
msgstr "保持 Crate 為最新版本"
```

File: po/zh-TW.po [hunk 11]
```diff
@@ -939,85 +924,85 @@ msgstr "錯誤處理"
...
 
 #: src/SUMMARY.md
 msgid "Bringing It Together - Exercise"
-msgstr ""
+msgstr "融會貫通 - 練習"
```
Comment (mingyc) [inline po/zh-TW.po:984]:
```suggestion
msgstr "融會貫通 - 練習題"
```

File: po/zh-TW.po [hunk 12]
```diff
@@ -939,85 +924,85 @@ msgstr "錯誤處理"
...
 
 #: src/SUMMARY.md src/exercises/chromium/solutions.md
 #, fuzzy
 msgid "Exercise Solutions"
-msgstr "解決方案"
+msgstr "練習的參考答案"
```
Comment (mingyc) [inline po/zh-TW.po:989]:
```suggestion
msgstr "練習題的參考答案"
```

File: po/zh-TW.po [hunk 13]
```diff
@@ -1610,134 +1595,132 @@ msgstr "課程架構"
...
 
 #: src/running-the-course/course-structure.md
 msgid "[Types and Values](../types-and-values.md) (45 minutes)"
-msgstr ""
+msgstr "[型別和值](../types-and-values.md) (5 分鐘)"
 
 #: src/running-the-course/course-structure.md
 msgid "[Control Flow Basics](../control-flow-basics.md) (40 minutes)"
-msgstr ""
+msgstr "[控制流程基本概念](../control-flow-basics.md) (40 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:1614]:
```suggestion
msgstr "[基本的控制流程概念](../control-flow-basics.md) (40 分鐘)"
```

File: po/zh-TW.po [hunk 14]
```diff
@@ -1610,134 +1595,132 @@ msgstr "課程架構"
...
-#, fuzzy
 msgid "[Pattern Matching](../pattern-matching.md) (1 hour)"
-msgstr ""
-"如要進一步瞭解 Rust 中的模式，請參閱「[模式比對](../pattern-matching.md)」。"
+msgstr "[模式比對](../pattern-matching.md) (1 小時)"
 
 #: src/running-the-course/course-structure.md
 msgid "[Methods and Traits](../methods-and-traits.md) (45 minutes)"
-msgstr ""
+msgstr "[方法和特徵](../methods-and-traits.md) (45 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:1646]:
```suggestion
msgstr "[方法 (methods) 和特徵 (traits)](../methods-and-traits.md) (45 分鐘)"
```

File: po/zh-TW.po [hunk 15]
```diff
@@ -1610,134 +1595,132 @@ msgstr "課程架構"
...
 
 #: src/running-the-course/course-structure.md
 msgid "[Generics](../generics.md) (40 minutes)"
-msgstr ""
+msgstr "[泛型](../generics.md) (40 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:1650]:
```suggestion
msgstr "[泛型 (generics)](../generics.md) (40 分鐘)"
```

File: po/zh-TW.po [hunk 16]
```diff
@@ -2430,37 +2433,37 @@ msgstr "模式配對：解構列舉、結構和陣列。"
...
 
 #: src/welcome-day-1.md
 msgid "[Welcome](./welcome-day-1.md) (5 minutes)"
-msgstr ""
+msgstr "[歡迎](./welcome-day-1.md) (5 分鐘)"
 
 #: src/welcome-day-1.md
 msgid "[Hello, World](./hello-world.md) (15 minutes)"
-msgstr ""
+msgstr "[Hello World](./hello-world.md) (15 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:2451]:
```suggestion
msgstr "[Hello, World](./hello-world.md) (15 分鐘)"
```

File: po/zh-TW.po [hunk 17]
```diff
@@ -2430,37 +2433,37 @@ msgstr "模式配對：解構列舉、結構和陣列。"
...
 
 #: src/welcome-day-1.md
 msgid "[Types and Values](./types-and-values.md) (45 minutes)"
-msgstr ""
+msgstr "[型別和值](./types-and-values.md) (45 分鐘)"
 
 #: src/welcome-day-1.md
 msgid "[Control Flow Basics](./control-flow-basics.md) (40 minutes)"
-msgstr ""
+msgstr "[控制流程基本概念](./control-flow-basics.md) (40 分鐘)"
```
Comment (mingyc) [inline po/zh-TW.po:2459]:
```suggestion
msgstr "[基本的控制流程概念](./control-flow-basics.md) (40 分鐘)"
```

**Review by mingyc — CHANGES_REQUESTED** [review]

File: po/zh-TW.po [hunk 18]
```diff
@@ -4109,6 +4196,9 @@ msgid ""
 "a reference to `point`, but `point` will be deallocated when the function "
 "returns, so this will not compile."
 msgstr ""
+"Rust 會追蹤所有參照的生命週期，確保其存留時間夠長。在安全的 Rust 中不會發生迷"
+"途參照。`x_axis` 會傳回對 `point` 的參照，但 `point` 會在函式傳回時釋放，因此"
```
Comment (mingyc) [inline po/zh-TW.po:4200]:
```suggestion
"途參照。`x_axis` 會傳回對 `point` 的參照，但在函式傳回時會釋放 `point` ，因此"
```

**Review by mingyc — CHANGES_REQUESTED** [review]
Sorry about multiple review cycles. This should be the last one for this PR.

File: po/zh-TW.po [hunk 19]
```diff
@@ -1773,8 +1755,8 @@ msgid ""
 "android/` directory into the root of your AOSP checkout. This will ensure "
 "that the Android build system sees the `Android.bp` files in `src/android/`."
 msgstr ""
-"您會需要進行 [Android 開放原始碼計畫檢查](https://source.android.com/docs/"
-"setup/download/downloading)。請檢查[課程存放區](https://github.com/google/"
+"您會需要進行 [AOSP 開放原始碼計畫檢查](https://source.android.com/docs/setup/"
+"download/downloading)。請檢查[課程存放區](https://github.com/google/"
 "comprehensive-rust)上相同的機器，並將 `src/android/` 目錄移至 Android 開放原"
```
[the remark this answers is not in the document]
Comment (henrif75) [reply po/zh-TW.po]: Hi @mingyc
Could you review this suggested change? I think GH mixed things up with the multiple suggestions here.
Thanks!

File: po/zh-TW.po [hunk 20]
```diff
@@ -1773,11 +1754,11 @@ msgid ""
...
 "that the Android build system sees the `Android.bp` files in `src/android/`."
 msgstr ""
-"您會需要進行 [Android 開放原始碼計畫檢查](https://source.android.com/docs/"
-"setup/download/downloading)。請檢查[課程存放區](https://github.com/google/"
-"comprehensive-rust)上相同的機器，並將 `src/android/` 目錄移至 Android 開放原"
-"始碼計畫檢查的根層級。這可確保 Android 建構系統在 `src/android/` 中看到 "
-"`Android.bp` 檔案。"
+"您會需要檢出 (checkout) [AOSP (Android 開放原始碼專案)](https://source."
+"android.com/docs/setup/download/downloading)。請檢查[課程存放區](https://"
+"comprehensive-rust)上相同的機器，並將 `src/android/` 目錄移至檢出的 Android "
```
Comment (mingyc) [inline po/zh-TW.po:1759]:
```suggestion
"comprehensive-rust)上相同的機器，並將 `src/android/` 目錄移至檢出的 AOSP "
```

File: po/zh-TW.po [hunk 21]
```diff
@@ -1773,11 +1754,11 @@ msgid ""
...
+"移至 AOSP 的根目錄。這可確保 Android 建構系統能夠在 `src/android/` 中 看到 "
```
Comment (mingyc) [inline po/zh-TW.po:1760]:
```suggestion
"根目錄下。這可確保 Android 建構系統能夠在 `src/android/` 中看到 "
```

**Review by mingyc — CHANGES_REQUESTED** [review]

File: po/zh-TW.po (hunk 19 above)
[the remark this answers is not in the document]
Comment (mingyc) [reply po/zh-TW.po]: Addressed in the latest comments. You might need to reformat again.

[the remark this answers is not in the document]
Comment (henrif75) [reply po/zh-TW.po]: Thanks!

**Review by mingyc — APPROVED** [review]

**Pull request merged**